profiler/options.go

```diff
@@ -253,3 +277,24 @@ func WithSite(site string) Option {
 		cfg.apiURL = u
 	}
 }
+
+// WithHTTPClient specifies the HTTP client to use when submitting profiles to Site.
+// In general, using this method is only necessary if you have need to customize the
+// transport layer, for instance when using a unix domain socket.
+func WithHTTPClient(client *http.Client) Option {
+	return func(cfg *config) {
+		cfg.httpClient = client
+	}
+}
+
+// WithUDS convenience, configures the client to dial a Unix Domain Socket
```

Comment:
This phrasing seems a bit off, perhaps we can adjust it to something similar like suggested above?